server/sonar-server/src/main/java/org/sonar/server/view/index/ViewIndexer.java

```diff
@@ -129,4 +136,34 @@ private void clearLookupCache(String viewUuid) {
     }
   }
 
+  @Override
+  public IndexingResult index(DbSession dbSession, Collection<EsQueueDto> items) {
+    if (items.isEmpty()) {
+      return new IndexingResult();
+    }
+
+    Set<String> logins = items
+      .stream()
+      .map(EsQueueDto::getDocId)
+      .collect(toHashSet(items.size()));
+
+    BulkIndexer bulkIndexer = newBulkIndexer(Size.REGULAR, new OneToOneResilientIndexingListener(dbClient, dbSession, items));
+    bulkIndexer.start();
+    logins.forEach(l -> bulkIndexer.addDeletion(INDEX_TYPE_VIEW, l));
```

Comment:
The resiliency mechanism should first try to index the view. If it does not exist then it should send a delete request to Elasticsearch.